### PR TITLE
Github actions integration

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
- Support for the SARIF upload to the github in plugins (gradle/maven)
- Updated actions
- Readme updates

### CODE REVIEW

**Comment by akuleshov7** [comment]
@petertrr just to save the example:
```
  "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
  "version": "2.1.0",
  "runs": [
    {
      "results": [
        {
          "level": "error",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "/home/runner/work/diktat/diktat/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter1/FileNaming.kt"
                },
                "region": {
                  "startColumn": 1,
                  "startLine": 32
                }
              }
            }
          ],
          "message": {
            "text": "[WRONG_INDENTATION] only spaces are allowed for indentation and each indentation should equal to 4 spaces (tabs are not allowed): expected 4 but was 14"
          },
          "ruleId": "diktat-ruleset:indentation"
        }
      ],
      "tool": {
        "driver": {
          "downloadUri": "https://github.com/pinterest/ktlint/releases/tag/0.43.2",
          "fullName": "ktlint",
          "informationUri": "https://github.com/pinterest/ktlint/",
          "language": "en",
          "name": "ktlint",
          "organization": "pinterest",
          "rules": [
          ],
          "version": "0.43.2"
        }
      }
    }
  ]
}
```

```
"uri": "/home/runner/work/diktat/diktat/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter1/FileNaming.kt"
```

**Comment by petertrr** [comment]
> "uri": "/home/runner/work/diktat/diktat/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter1/FileNaming.kt"

According to this code from ktlin-sarif-reporter
https://github.com/pinterest/ktlint/blob/10e8a9bedc9f9d25a2eadfe08d935a3ad25dee83/ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt#L50-L57
uri is an absolute path if proeprty `user.home` is unset.

**Comment by akuleshov7** [comment]
https://sarifweb.azurewebsites.net/Validation

**Comment by akuleshov7** [comment]
image

**Comment by petertrr** [comment]
> According to this code from ktlin-sarif-reporter https://github.com/pinterest/ktlint/blob/10e8a9bedc9f9d25a2eadfe08d935a3ad25dee83/ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt#L50-L57 uri is an absolute path if proeprty `user.home` is unset.

Actually, we don't call `reporterImpl.beforeAll()` in maven plugin, so `workingDirectory` is unset an reporter uses absolute paths.

**Comment by akuleshov7** [comment]
> > According to this code from ktlin-sarif-reporter https://github.com/pinterest/ktlint/blob/10e8a9bedc9f9d25a2eadfe08d935a3ad25dee83/ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt#L50-L57 uri is an absolute path if proeprty `user.home` is unset.
> 
> Actually, we don't call `reporterImpl.beforeAll()` in maven plugin, so `workingDirectory` is unset an reporter uses absolute paths.

Yeah, I already saw it yesterday when I set the environment property

**Comment by akuleshov7** [comment]
%SRCROOT%, in baseApi now, very strange
```
        "%SRCROOT%": {
          "uri": "file:///home/runner/work/diktat/diktat/diktat-rules/"
```

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter1/FileNaming.kt [hunk 1]
```diff
@@ -29,7 +29,7 @@ import java.io.File
 class FileNaming(configRules: List<RulesConfig>) : DiktatRule(
     "file-naming",
     configRules,
-    listOf(FILE_NAME_INCORRECT, FILE_NAME_MATCH_CLASS)
+              listOf(FILE_NAME_INCORRECT, FILE_NAME_MATCH_CLASS)
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter1/FileNaming.kt:32]: hui pidr test

File: diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt [hunk 2]
```diff
@@ -157,22 +157,37 @@ open class DiktatJavaExecTaskBase @Inject constructor(
         // appending the flag with the reporter
         setReporter(diktatExtension, flag)
 
-        if (diktatExtension.output.isNotEmpty()) {
-            flag.append(",output=${diktatExtension.output}")
+        val outFlag = when {
+            // githubActions should have higher priority than a custom input
+            diktatExtension.githubActions -> ",output=${project.projectDir}/${project.name}"
```
Comment (petertrr) [inline diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:162]:
```suggestion
            diktatExtension.githubActions -> ",output=${project.buildDir}/${project.name}"
```
I think it's better to keep repo clean

**Review by petertrr — APPROVED** [review]

File: diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt (hunk 2 above)
Comment (akuleshov7) [reply diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:162]: By for now, let's create grabage. As it happens in separate action - it should not affect anyone. But I feel more safe :)

**Pull request merged**